general/Random.h

```diff
@@ -0,0 +1,147 @@
+/**
+ * \section License
+ * This file is part of JuPedSim.
+ *
+ * JuPedSim is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * JuPedSim is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with JuPedSim. If not, see <http://www.gnu.org/licenses/>.
+ **/
+#pragma once
+
+#include <random>
+#include <stdexcept>
+
+/**
+ * Toolset for creating random numbers
+ */
+namespace Random
+{
+namespace
+{
+/**
+ * Random engine used for creating the random numbers
+ */
+std::mt19937 _randomEngine;
+
+/**
+ * Flag if a seed was properly set
+ */
+bool _init = false;
+} // namespace
+
+/**
+ * Seed the random number generator
+ * @param seed seed used for the random number generator
+ */
+void Setup(int seed)
+{
+    _randomEngine = std::mt19937(seed);
+    _init         = true;
+}
+
+/**
+ * Returns a random number of a normal distribution with mean (\p mean) and standard deviation (\p stddev).
+ * @tparam RealType
+ * @param mean mean of the normal distribution
+ * @param stddev standard deviation of the normal distribution
+ * @return random number of a normal distribution with given parameters
+ */
+template <class RealType = double>
```

Comment:
I would prefer `typename` instead of `class`. Especially since RealType is not a class in most cases.